python/tutorials/11-programmatic-dependent-launch.py

```diff
@@ -0,0 +1,265 @@
+        launch_pdl=launch_pdl,  # launch kernel with PDL flag set enabled
+    )
+    return c
+
+
+def validate(M, N, K, dtype):
+    a = torch.randn((M, K), device="cuda", dtype=torch.float16).to(dtype)
+    b = torch.randn((K, N), device="cuda", dtype=torch.float16).to(dtype)
+    b = b.T.contiguous()
+
+    torch_result = torch.matmul(a, b.T)
+    matmul_result = matmul(a, b)
+
+    torch_vs_matmul = "✅" if torch.allclose(torch_result.to(torch.float16), matmul_result.to(torch.float16),
+                                            atol=1.0) else "❌"
+    print(f"M={M}, N={N}, K={K} verification naive vs: ", end="")
+    print(f"matmul: {torch_vs_matmul} ", end="")
+    print()
+
+
+TORCH_HAS_FP8 = hasattr(torch, 'float8_e5m2')
+
+
+@triton.testing.perf_report(
+    triton.testing.Benchmark(
+        x_names=["K"],
+        x_vals=range(512, 4096 + 1, 512),
+        x_log=False,
+        line_arg="provider",
+        line_vals=["pdl-fp16", "fp16"] + (["pdl-fp8", "fp8"] if TORCH_HAS_FP8 else []),
+        line_names=["PDL [FP16]", "No PDL [FP16]"] + (["PDL [FP8]", "No PDL [FP8]"] if TORCH_HAS_FP8 else []),
+        styles=[("red", "-"), ("blue", "-"), ("green", "-"), ("orange", "-")],
+        ylabel="TFLOPS",
+        plot_name="pdl-performance",
+        args={},
+    ))
+def benchmark(K, provider, M, N):
+
+    tflops = (lambda ms: (2.0 * M * N * K) * 1e-12 / (ms * 1e-3))
+
+    quantiles = [0.5, 0.2, 0.8]
+    if "fp8" in provider:
+        a = torch.randn((M, K), device="cuda", dtype=torch.float16).to(torch.float8_e5m2)
+        b = torch.randn((K, N), device="cuda", dtype=torch.float16).to(torch.float8_e5m2)
+        b = b.T.contiguous()
+        if "pdl" in provider:
+            ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
+                lambda: matmul(a, b),
+                quantiles=quantiles,
+                rep=1000,
+            )
+            return tflops(ms), tflops(max_ms), tflops(min_ms)
+        else:
+            ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
+                lambda: matmul(a, b, False),
+                quantiles=quantiles,
+                rep=1000,
+            )
+            return tflops(ms), tflops(max_ms), tflops(min_ms)
+    else:
+        a = torch.randn((M, K), device="cuda", dtype=torch.float16)
+        b = torch.randn((K, N), device="cuda", dtype=torch.float16)
+        b = b.T.contiguous()
+        if "pdl" in provider:
+            ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
+                lambda: matmul(a, b),
+                quantiles=quantiles,
+                rep=1000,
+            )
+            return tflops(ms), tflops(max_ms), tflops(min_ms)
+        else:
+            ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
+                lambda: matmul(a, b, False),
+                quantiles=quantiles,
+                rep=1000,
+            )
+            return tflops(ms), tflops(max_ms), tflops(min_ms)
```

Comment:
NIT: This is unnecessarily verbose
```suggestion
    dtype = torch.float8_e5m2 if "fp8" in provider else torch.float16
    a = torch.randn((M, K), device="cuda", dtype=torch.float16).to(dtype)
    b = torch.randn((K, N), device="cuda", dtype=torch.float16).to(dtype)
    b = b.T.contiguous()
    
    use_pdl = "pdl" in provider:
    ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
        lambda: matmul(a, b, use_pdl),
        quantiles=quantiles,
        rep=1000,
    )
    return tflops(ms), tflops(max_ms), tflops(min_ms)
```